internal/pluginshared/cloudclient.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package pluginshared
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/hashicorp/terraform/internal/httpclient"
+	"github.com/hashicorp/terraform/internal/logging"
+)
+
+// NewCloudPluginClient creates a new client for downloading and verifying
+// terraform-cloudplugin archives
+func NewCloudPluginClient(ctx context.Context, serviceURL *url.URL) (*BasePluginClient, error) {
+	httpClient := httpclient.New()
+	httpClient.Timeout = defaultRequestTimeout
+
+	retryableClient := retryablehttp.NewClient()
+	retryableClient.HTTPClient = httpClient
+	retryableClient.RetryMax = 3
+	retryableClient.RequestLogHook = requestLogHook
+	retryableClient.Logger = logging.HCLogger()
+
+	client := BasePluginClient{
+		ctx:        ctx,
+		serviceURL: serviceURL,
+		httpClient: retryableClient,
+		pluginName: "cloudplugin",
```

Comment:
It looks like you are scrubbing the term "cloudplugin" from this package, which I think is a good idea. But it's not clear to me if you went as far as you could go or if you just missed a few of them. This is just one example.

IMO, it's OK to remove or manipulate all cloudplugin features from terraform and shift those over to stacks

Reply:
I went about the rearrangement by creating a `base client` so that each plugin could use an instance of the base client, that's why there is still both `cloudclient` and `stacksclient` in the package.